app/assets/javascripts/discourse/tests/acceptance/new-topic-test.js

```diff
@@ -17,26 +13,53 @@ acceptance("New Topic - Anonymous", function () {
 
 acceptance("New Topic - Authenticated", function (needs) {
   needs.user();
-  test("accessing new-topic route when logged in", async function (assert) {
+
+  test("accessing new-topic route", async function (assert) {
+    await visit("/c/1");
+
+    try {
+      await visit("/new-topic");
+    } catch (error) {
+      assert.strictEqual(
+        error.message,
+        "TransitionAborted",
+        "it aborts the transition"
+      );
+    }
+
+    assert.strictEqual(currentURL(), "/c/1");
+
+    await waitUntil(
+      () => {
+        return exists(".composer-fields");
+      },
+      { timeout: 5000 }
+    );
```

Comment:
This is necessary because the composer is opened in the next runloop so we have no way to "await" on it.

Reply:
you can use `await waitFor(".composer-fields", { timeout: 5000 });` instead here 😃